File: packages/stripe/src/HeliumStripeSdkModule.ts
import { NativeModule, requireNativeModule } from "expo";
import { Platform } from "react-native";

declare class HeliumStripeSdkModule extends NativeModule {
  initializeStripe(config: Record<string, any>): void;

  setUserIdAndSyncStripeIfNeeded(userId: string): void;

  resetStripeEntitlements(clearUserId: boolean): void;

  createStripePortalSession(returnUrl: string): Promise<string>;

  hasActiveStripeEntitlement(): Promise<boolean>;
}

// Only resolve the native module on iOS. On Android there is no native
// HeliumStripeSdk module — all exported functions guard on Platform.OS
// before touching this reference.
export default (Platform.OS === "ios"
  ? requireNativeModule<HeliumStripeSdkModule>("HeliumStripeSdk")
  : null) as HeliumStripeSdkModule;

File: packages/stripe/android/src/main/java/expo/modules/stripesdk/HeliumStripeSdkModule.kt
package expo.modules.stripesdk

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class HeliumStripeSdkModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("HeliumStripeSdk")

        // No-op implementations — Stripe One Tap is iOS-only for now.
        // All exported JS functions guard on Platform.OS before calling these.

        Function("initializeStripe") { _: Map<String, Any> -> }

        Function("setUserIdAndSyncStripeIfNeeded") { _: String -> }
    }
}
